docs/styleguides/python.md

```diff
@@ -0,0 +1,173 @@
+                    masters: List[str],
+                    public_masters: List[str],
+                    slaves: List[str],
+                    public_slaves: List[str],
+                    default_os_user: str) -> DcosApiSession:
+
+    * Reviewers will request the author to write types for the method parameters.
+    * We will start running mypy type checker as part of build process.
+
+* Shebang Rule.
+    * We will consistently use the shebang line in our scripts.
+        * `#!/usr/bin/env python3`
+        OR
+        * `#!/usr/bin/env python2` in our scripts.
+
+* Use the `format` method
+    * Prefer `'{named}'.format` will be over `{}.format`.
+        * Rationale: If the number of arguments are long. `{named}.format` helps in readability.
+    * Use your best judgement to decide between + and % (or format) when applicable.
+
+* TODO Comment Format
+    * Follow the rules on [when to write TODO comments](https://google.github.io/styleguide/pyguide.html?showone=TODO_Comments#TODO_Comments)
+    * Format required will be `TODO (active-developer) [JIRA] - Brief one line description`
+    * A plugin can be used to enforce the TODO format requirement.
+
+* [Comments and Docstrings](https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments).
+    * Follow the base style guidelines to adopt the style that explains arguments, return value and exceptions.
+
+```
+class DcosApiSession(ARNodeApiClientMixin, ApiClientSession):
+    def __init__(
+            self,
+            dcos_url: str,
+            masters: list,
+            public_masters: list,
+            slaves: list,
+            public_slaves: list,
+            default_os_user: str,
+            auth_user: Optional[DcosUser]):
+
+        """Proxy class for DC/OS clusters.
+
+        This class provides facilities to interact with the DCOS Cluster launched in the integration tests.
+
+
+        Args:
+            dcos_url: address for the DC/OS web UI.
+            masters: list of Mesos master advertised IP addresses.
+            public_masters: list of Mesos master IP addresses routable from
+                the local host.
+            slaves: list of Mesos slave/agent advertised IP addresses.
+            default_os_user: default user that marathon/metronome will launch tasks under
+            auth_user: use this user's auth for all requests
+                Note: user must be authenticated explicitly or call self.wait_for_dcos()
+        Returns:
+            None
+        Raises:
+            AssertionErrors on empty values.
+            Exception when a given node is not recognized within the DCOS Cluster.
+        """
+```
+
+* We can use sphinx documentation generator. Tools like [sphinx-napolean plugin](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/) can be used to automatically generate api documentation with the above chosen convention.
+
+#### Style Enforcements.
+
+* Editor configurations can help with indentation and formatting.
```

Comment:
I'd rather remove this statement.  I'd not like to suggest what plugins that developers should use. Instead, the style guide should state the requirements and the developers can configure their editors to meet those requirements.